std/string.d

```diff
@@ -2033,6 +2040,211 @@ S[] splitLines(S)(S s, in KeepTerminator keepTerm = KeepTerminator.no) @safe pur
+            _input = input;
+        }
+
+        static if (isInfinite!Range)
+        {
+            enum bool empty = false;
+        }
+        else
+        {
+            @property bool empty()
+            {
+                return iStart == _unComputed && iNext == _input.length;
+            }
+        }
+
+        @property Range front()
+        {
+            if (iStart == _unComputed)
+            {
+                iStart = iNext;
+              Loop:
+                for (IndexType i = iNext; true; ++i)
+                {
+                    if (i == _input.length)
+                    {
+                        iEnd = i;
+                        iNext = i;
+                        break Loop;
+                    }
+                    switch (_input[i])
+                    {
+                        case '\n':
+                            iEnd = i + (keepTerm == KeepTerminator.yes);
+                            iNext = i + 1;
+                            break Loop;
+
+                        case '\r':
+                            if (i + 1 < _input.length && _input[i + 1] == '\n')
+                            {
+                                iEnd = i + (keepTerm == KeepTerminator.yes) * 2;
+                                iNext = i + 2;
+                                break Loop;
+                            }
+                            else
+                            {
+                                goto case '\n';
+                            }
+
+                        static if (_input[i].sizeof == 1)
+                        {
+                            /* Manually decode:
+                             *  lineSep is E2 80 A8
+                             *  paraSep is E2 80 A9
+                             */
+                            case 0xE2:
+                                if (i + 2 < _input.length &&
+                                    _input[i + 1] == 0x80 &&
+                                    (_input[i + 2] == 0xA8 || _input[i + 2] == 0xA9)
+                                   )
+                                {
+                                    iEnd = i + (keepTerm == KeepTerminator.yes) * 3;
+                                    iNext = i + 3;
+                                    break Loop;
+                                }
+                                else
+                                    goto default;
+                        }
+                        else
+                        {
+                            case lineSep:
+                            case paraSep:
+                                goto case '\n';
+                        }
+
+                        default:
+                            break;
+                    }
+                }
+            }
+            return _input[iStart .. iEnd];
+        }
+
+        void popFront()
+        {
+            if (iStart == _unComputed)
+            {
+                assert(!empty);
+                front();
```

Comment:
1. that's what it does
2. that's what it does
3. not right, as popFront can be called multiple times and must advance, Hence it must contain almost all of the logic in front(). In order to avoid copy/pasta bugs, I just called front.